ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/rule-violations/index.tsx

```diff
@@ -18,26 +18,236 @@
  */
 
 import { createFileRoute } from '@tanstack/react-router';
+import {
+  ColumnDef,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getGroupedRowModel,
+  getPaginationRowModel,
+  useReactTable,
+} from '@tanstack/react-table';
+import { ChevronRight } from 'lucide-react';
+import { useMemo } from 'react';
 
+import { prefetchUseRepositoriesServiceGetOrtRunByIndex } from '@/api/queries/prefetch';
+import {
+  useRepositoriesServiceGetOrtRunByIndexSuspense,
+  useRuleViolationsServiceGetRuleViolationsByRunIdSuspense,
+} from '@/api/queries/suspense';
+import { RuleViolationWithIdentifier } from '@/api/requests';
+import { DataTable } from '@/components/data-table/data-table';
 import { LoadingIndicator } from '@/components/loading-indicator';
+import { Badge } from '@/components/ui/badge';
+import { Button } from '@/components/ui/button';
 import {
   Card,
   CardContent,
   CardDescription,
   CardHeader,
   CardTitle,
 } from '@/components/ui/card';
-import { Label } from '@/components/ui/label';
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@/components/ui/tooltip';
+import { getRuleViolationSeverityBackgroundColor } from '@/helpers/get-status-class';
+import { paginationSchema, tableGroupingSchema } from '@/schemas';
+
+const defaultPageSize = 10;
+
+const columns: ColumnDef<RuleViolationWithIdentifier>[] = [
+  {
+    accessorFn: ({ ruleViolation }) => ruleViolation.severity,
+    header: 'Severity',
+    cell: ({ row }) => {
+      return (
+        <Badge
+          className={`${getRuleViolationSeverityBackgroundColor(row.original.ruleViolation.severity)}`}
+        >
+          {row.original.ruleViolation.severity}
+        </Badge>
+      );
+    },
+  },
+  {
+    accessorFn: ({ identifier: { name, namespace, type, version } }) => {
+      return `${type ? type.concat(':') : ''}${namespace ? namespace.concat('/') : ''}${name ? name : ''}${version ? '@'.concat(version) : ''}`;
+    },
+    header: 'Package',
+    cell: ({ getValue }) => {
+      // TypeScript gets confused, but type casting to string is safe here,
+      // because the accessor function returns a string.
+      return <div className='font-semibold'>{getValue() as string}</div>;
+    },
+  },
+  {
+    accessorFn: ({ ruleViolation }) => ruleViolation.rule,
+    header: 'Rule',
+    cell: ({ row }) => (
+      <Badge className='whitespace-nowrap bg-gray-400'>
+        {row.original.ruleViolation.rule}
+      </Badge>
+    ),
+  },
+  {
+    id: 'moreInfo',
+    header: () => null,
+    enableGrouping: false,
+    size: 50,
+    cell: ({ row }) => (
+      <TooltipProvider>
```

Comment:
I think it does make sense to rather use one TooltipProvider for the whole table, as this creates a provider for each row, which can be a little inefficient.